src/components/accounts/RecoverAccountSeedPhrase.js

```diff
@@ -26,36 +25,48 @@ class RecoverAccountSeedPhrase extends Component {
 
     componentDidMount = () => {}
 
-    componentWillUnmount = () => {
-        this.props.clear()
-    }
-
     handleChange = (e, { name, value }) => {
         this.setState(() => ({
             [name]: value
         }))
+
+        this.props.clear()
     }
 
     handleSubmit = () => {
-
         if (!this.isLegit) {
             return false
         }
 
-        const accountId = this.state.accountId
-        this.props.recoverAccountSeedPhrase(this.state.seedPhrase, accountId)
+        this.props.recoverAccountSeedPhrase(this.state.seedPhrase)
             .then(({ error }) => {
                 if (error) return
                 this.props.refreshAccount()
                 this.props.redirectToApp()
             })
     }
 
+    checkInvalidSeedPhrase = () => this.props.requestStatus.errorMessage && this.props.requestStatus.errorMessage.includes('Cannot find matching public key')
```

Comment:
nit: let's name this `isSeedPhraseValid` (and reverse meaning) as it's not clear from `check` prefix whether method will do something or just return `true` / `false`. 

Is there any way to avoid depending on specific wording of message in thrown error?

Reply:
> Is there any way to avoid depending on specific wording of message in thrown error?

We can add error code to error and change `readyStatePromise` to return this code as well - then we would be able to pass it through `requestResultReducer`, and check code instead of a message. This code can be Translate id, so it will be easy to show it.

Let me know what you think. If you are OK with that let me handle it in separate PR.

Reply:
@marcinbodnar I like this idea, but I don't understand how do you want to modify `readyStatePromise`. Shouldn't it just passthrough `Error` that was thrown, so error code just needs to be part of `Error`?

Reply:
>Shouldn't it just passthrough Error that was thrown, so error code just needs to be part of Error?

@vgrichina you're right, we don't need to modify `readyStatePromise`.